src/client/Client.cc

```diff
@@ -10679,6 +10679,7 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
   req->head.args.setxattr.flags = xattr_flags;
 
   bufferlist bl;
+  assert (value != NULL || size != 0);
```

Comment:
No. It should be `assert(value || size == 0)`. A zero-length value is permitted. Either `value` must be non-NULL (so any value of size is permitted) OR `size` must be 0 (meaning `value` may be NULL).

Perhaps that will still raise a coverity error but that is the correct assertion.